src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -86,9 +86,14 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
 
         try {
 
-            CompanyProfileDocument cpDocument =
-                    companyProfileRepository.findById(companyNumber).get();
+            Optional<CompanyProfileDocument> cpDocumentOptional =
+                    companyProfileRepository.findById(companyNumber);
+
+            cpDocumentOptional.orElseThrow(() -> new IllegalArgumentException(
+                    String.format("No company profile with company number %s found",
+                    companyNumber)));
```

Comment:
I believe this needs to be thrown as a `BadRequestException` as it was originally (mainly due to the status code expected)